### PR TITLE
Move initialization helpers to ParallelAlgorithms

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Moves `Initialization::merge_into_databox` and `Initialization::Actions::RemoveOptionsAndTerminatePhase` from `Evolution` to `ParallelAlgorithms`. This is in preparation for generalising the initialisers.

### Upgrade instructions

- For `Initialization::merge_into_databox` and `Initialization::Actions::RemoveOptionsAndTerminatePhase` you should now include the headers from `ParallelAlgorithms` instead of `Evolution`.

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: tests/Unit/ParallelAlgorithms/Initialization/Actions/Test_RemoveOptionsAndTerminatePhase.cpp [hunk 1]
```diff
@@ -24,6 +23,11 @@
 namespace {
 struct TemporalId {};
 
+struct InitialTime : db::SimpleTag {
```
Comment (nilsdeppe) [inline tests/Unit/ParallelAlgorithms/Initialization/Actions/Test_RemoveOptionsAndTerminatePhase.cpp:26]: How is this related? Please remove it or put it in a separate commit. "Move" commits should literally only do the move.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

File: tests/Unit/ParallelAlgorithms/Initialization/Actions/Test_RemoveOptionsAndTerminatePhase.cpp (hunk 1 above)
Comment (nilsvu) [reply tests/Unit/ParallelAlgorithms/Initialization/Actions/Test_RemoveOptionsAndTerminatePhase.cpp:26]: Since I’m moving this out of `Evolution` I remove the dependence on evolution tags. Are you sure you prefer this in a separate commit?

Comment (nilsdeppe) [reply tests/Unit/ParallelAlgorithms/Initialization/Actions/Test_RemoveOptionsAndTerminatePhase.cpp:26]: Yes, always. "Move" commits should be completely trivial to review, otherwise it's not a move, you're actually changing code.

Comment (nilsvu) [reply tests/Unit/ParallelAlgorithms/Initialization/Actions/Test_RemoveOptionsAndTerminatePhase.cpp:26]: ok, did the split

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]
